js/editor-controller.js

```diff
@@ -0,0 +1,229 @@
+/**
+ * The Editor can be controlled by remote hosts that wrap the editor with
+ * an iframe or in some kind of webview.
+ */
+'use strict';
+
+/**
+ * @returns An object with the fs wrapper.
+ */
+var EditorController = function() {
+    var _returnObj = {};
+
+    var _editorActions = null;
+    var _controllerHost = null;
+
+    /** Constants used for iframe messaging */
+    var CONTROLLER_MESSAGING = Object.freeze({
+        // Embed editor host type
+        type: 'pyeditor',
+        // Embed editor messaging actions
+        actions: {
+            workspacesync: 'workspacesync',
+            workspacesave: 'workspacesave',
+            workspaceloaded: 'workspaceloaded',
+            importproject: 'importproject',
+            loadhex: 'loadhex',
+            loadfile: 'loadfile',
+            mobilemode: 'mobilemode',
```

Comment:
Renamed downloadmode to mobilemode. This message is used to change the editor buttons to send messages to the apps.

This is done on request to allow other applications to embed the editor and use the protocol without automatically changing the editor itself.

Reply:
@microbit-sam @martinwork I assume this is fine?

Reply:
Yeah that's fine by me